frontends/pytorch/csrc/c10_dispatch/python_bindings.cpp

```diff
@@ -0,0 +1,129 @@
+public:
+  using CallbackTy = std::function<void(const c10::OperatorHandle &)>;
+  LambdaOpRegistrationListener(CallbackTy callback)
+      : callback(std::move(callback)) {}
+  void onOperatorRegistered(const c10::OperatorHandle &op) override {
+    callback(op);
+  }
+  void onOperatorDeregistered(const c10::OperatorHandle &op) override {}
+
+private:
+  CallbackTy callback;
+};
+
+py::list GetRegisteredOps() {
+  py::list results;
+  c10::Dispatcher &dispatcher = c10::Dispatcher::singleton();
+  auto listener = std::make_unique<LambdaOpRegistrationListener>(
+      [&](const c10::OperatorHandle &op) -> void {
+        if (!op.hasSchema()) {
+          // Legacy?
+          return;
+        }
+
+        py::dict record;
+        {
+          py::tuple name(2);
+          name[0] = op.operator_name().name;
+          name[1] = op.operator_name().overload_name;
+          record["name"] = std::move(name);
+        }
+
+        auto &schema = op.schema();
+        record["is_vararg"] = schema.is_vararg();
+        record["is_varret"] = schema.is_varret();
+        record["is_mutable"] = schema.is_mutable();
+
+        py::list arguments;
+        py::list returns;
+        auto addArgument = [](py::list &container, const c10::Argument &arg) {
+          py::dict argRecord;
+          argRecord["name"] = arg.name();
+          argRecord["type"] = arg.type()->str();
+          argRecord["pytype"] = arg.type()->annotation_str();
+          if (arg.N())
+            argRecord["N"] = *arg.N();
+          // TODO: If the default value becomes useful, switch on it and return
+          // a real value, not just a string print.
+          if (arg.default_value()) {
+            std::stringstream sout;
+            sout << *arg.default_value();
+            argRecord["default_debug"] = sout.str();
+          }
+          if (arg.alias_info()) {
+            py::dict aliasInfo;
+            py::list before;
+            py::list after;
+            for (auto &symbol : arg.alias_info()->beforeSets()) {
+              before.append(std::string(symbol.toQualString()));
+            }
+            for (auto &symbol : arg.alias_info()->afterSets()) {
+              after.append(std::string(symbol.toQualString()));
+            }
+            aliasInfo["before"] = std::move(before);
+            aliasInfo["after"] = std::move(after);
+            argRecord["alias_info"] = std::move(aliasInfo);
+          }
+
+          container.append(std::move(argRecord));
+        };
+        for (auto &argument : schema.arguments()) {
+          addArgument(arguments, argument);
+        }
+        for (auto &returnArg : schema.returns()) {
+          addArgument(returns, returnArg);
+        }
+        record["arguments"] = std::move(arguments);
+        record["returns"] = std::move(returns);
+        results.append(std::move(record));
+      });
+  { dispatcher.addRegistrationListener(std::move(listener)); }
```

Comment:
I find these braces hard to understand. Why are they needed?

Reply:
Ugh. Added this note:

```
  // Note: addRegistrationListener reports all currently registered ops
  // during the call and then incrementally reports newer ops until the RAII
  // return value is destroyed. Since we only want the current, surround in
  // a block so it immediately unregisters.
```

I didn't trace through any asynchronicity that may be involved and just wanted to ensure there was an unregistration prior to manipulating further.